Sage/carbon
# feat(menu-item): add `submenuMaxWidth` prop to allow setting of `max-width` on submenus

### Proposed behaviour


Adds `submenuMaxWidth` prop to allow setting of `max-width` on the `submenu` child of a `MenuItem`. When the prop is set any submenu items will wrap so they do not overflow the bounds of the container.

![image](https://github.com/user-attachments/assets/7efad75d-f22c-46c8-8b5e-746f2e25aaf2)

Replaces display inline-block with flex and removes height and line-height as it makes the spacing incorrect when the items wrap

Updates how padding is applied so it is set on the button/anchor elements rather than the wrapper
fix #6889

![image](https://github.com/user-attachments/assets/b0ecc832-81f6-47bc-b478-4a6e4df1de50)

### Current behaviour


There is no support for setting max width on the submenu elements, child menu-items do not wrap.
Height and line-height are 40px and display is inline-block
Padding is applied to the wrapper element rather than the button/anchor element so that even when it is set to zero the underlying element's padding means there's no visual update

### Checklist




#### QA


### Additional context



### Testing instructions


`menu--truncation-and-submenu-width` added for testing/demoing of updates
`menu-test--menu-with-submenu-custom-padding` added for padding regression

- As I have had to make changes to how the items display I suggest that `Menu` is thoroughly tested to ensure there have been no visual regressions
